### ISSUE TITLE
Add pre-commit hooks

### ISSUE THREAD

**Issue by Strift** [meilisearch/documentation#2718]
It often happens that PR are submitted (by me 🤡) without running the linter first.

Adding a pre-commit hook to automatically run the linter before creating a commit can help with ensure that this is done and doesn't waste reviewers time. 

(It can still be possible to commit bypass this protection if needed when doing quick & dirty stuff, the CI will still run for PRs)

**Comment by guimachiavelli** [comment]
The docs team had this discussion a long time ago and I believe @curquiza was against running linters on actions because this might dissuade contributors from submitting PRs.

**Comment by curquiza** [comment]
Yeah I'm not a big fan of imposed pre-commit hook. It definitely can prevent people from submitting. It prevented me in the past for example 😅
We would rather have a contribution with linting issue (we can update ourselves) rather than no contribution at all. The linting problem should be visible in the CI associated to the PR. It's already the case right?

But only my opinion, I had bad experiences with pre-commit hook when trying to contribute to other opensource repositories.

**Comment by curquiza** [comment]
Also, I think pre-commit hook can be set up individually (so you can set it up on your side @Strift) without imposing it to everyone

**Comment by Strift** [comment]
I think it's fine to have it in CI only for now :)
